packages/jsii/lib/compiler.ts

```diff
@@ -72,28 +72,79 @@ export class Compiler implements Emitter {
   }
 
   /**
-     * Compiles the configured program.
-     *
-     * @param files can be specified to override the standard source code location logic. Useful for example when testing "negatives".
-     */
+   * Adds a listener for the 'result' event that is called with EmitResult.
+   *
+   * @param event    the 'result' event
+   * @param listener the listener to be called with the EmitResult.
+   */
+  public on(event: 'result', listener: (result: EmitResult) => any): this;
+  /**
+   * Adds a listener for the 'statusChanged' event that is called with a Diagnostic whenever the
+   * status of the current watch process changes.
+   *
+   * @param event    the 'statusChanged' event
+   * @param listener the listener to be called with the Diagnostic.
+   */
+  public on(event: 'statusChanged', listener: (message: Diagnostic) => any): this;
+  public on(event: string, listener: (...args: any[]) => any): this {
+    this.eventEmitter.on(event, listener);
+    return this;
+  }
+
+  /**
+   * Adds a once-only listener for the 'result' event that is called with EmitResult.
+   *
+   * @param event    the 'result' event
+   * @param listener the listener to be called with the EmitResult.
+   */
+  public once(event: 'result', listener: (result: EmitResult) => any): this;
+  /**
+   * Adds a once-only listener for the 'statusChanged' event that is called with a Diagnostic
+   * whenever the status of the current watch process changes.
+   *
+   * @param event    the 'statusChanged' event
+   * @param listener the listener to be called with the Diagnostic.
+   */
+  public once(event: 'statusChanged', listener: (message: Diagnostic) => any): this;
+  public once(event: string, listener: (...args: any[]) => any): this {
+    this.eventEmitter.once(event, listener);
+    return this;
+  }
+
+  /**
+   * Compiles the configured program.
+   *
+   * @param files can be specified to override the standard source code location logic. Useful for example when testing "negatives".
+   */
   public async emit(...files: string[]): Promise<EmitResult | never> {
+    await this._prepareForBuild(...files);
+    return this._buildOnce();
+
+  }
+
+  /**
+   * Watches for file changes and dynamically re-compiles whenever needed.
+   */
+  public async watch(): Promise<Watch> {
+    await this._prepareForBuild();
+    return this._startWatch();
```

Comment:
Why do we need `_startWatch` as a separate method?

Reply:
Good point. It's probably a remnant of multiple successive refactors that gradually thinned the `watch` method until nothing was left in it :)